containous/maesh
# Update smi-sdk-go to v0.4

## What does this PR do?

This PR updates `smi-sdk-go` to [v0.4](https://github.com/servicemeshinterface/smi-sdk-go/releases/tag/v0.4.0). This new version bring support for [`smi-spec` v0.5](https://github.com/servicemeshinterface/smi-spec/blob/master/SPEC_LATEST_STABLE.md)

The SMI CRD versions haven't been updated. A [dedicated task](https://github.com/containous/maesh/issues/598) will handle this.

Fixes #689, #602

### How to test it

* make test
* make test-integration
* Install Maesh on a fresh cluster, apply a TrafficTarget, TrafficSplit and some services and make sure it still behave correctly.

## Additional Notes

`smi-sdk-go` v0.4 relies on k8s v0.18.0. Traefik version had to be updated to v2.3.0-rc2. A new PR will be opened once the v2.3.0 GA will be available.

Co-authored-by: Kevin Pollet pollet.kevin@gmail.com
